### PR TITLE
Move ChannelCount triggers to using an insert/delete approach

### PR DESCRIPTION
**Pull request by nicpottier**
Switches to using INSERT and DELETE operations and squashing counts every ~100 inserts or so.

### CODE REVIEW

File: temba/channels/migrations/0021_auto_20150803_1857.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ]
+
+    def install_updated_increment(apps, schema_editor):
+        #language=SQL
+        update_trigger = """
+            CREATE OR REPLACE FUNCTION temba_decrement_channelcount(_channel_id INTEGER, _count_type VARCHAR(2), _count_day DATE) RETURNS VOID AS $$
+              BEGIN
+                INSERT INTO channels_channelcount("channel_id", "count_type", "day", "count")
+                  VALUES(_channel_id, _count_type, _count_day, -1);
+                PERFORM temba_maybe_squash_channelcount(_channel_id, _count_type, _count_day);
```
Comment (ericnewcomer) [inline temba/channels/migrations/0021_auto_20150803_1857.py]: ...memories!

Comment (nicpottier) [inline temba/channels/migrations/0021_auto_20150803_1857.py]: Good name no?

Comment (ericnewcomer) [inline temba/channels/migrations/0021_auto_20150803_1857.py]: yes, it is indeed perfect. and it makes me smile inside.

File: temba/channels/models.py [hunk 2]
```diff
@@ -1842,9 +1842,18 @@ class ChannelCount(models.Model):
     count = models.IntegerField(default=0,
                                 help_text=_("The count of messages on this day and type"))
 
-    class Meta:
-        unique_together = ('channel', 'day', 'count_type')
+    @classmethod
+    def get_day_count(cls, channel, count_type, day):
+        count = ChannelCount.objects.filter(channel=channel, count_type=count_type, day=day).\
+          order_by('day', 'count_type').aggregate(count_sum=Sum('count'))
+
```
Comment (ericnewcomer) [inline temba/channels/models.py]: I guess in this particular case it's really not ugly at all on the python side

Comment (nicpottier) [inline temba/channels/models.py]: Ya, honestly I prefer less magic here.. it doesn't hide that there could be more than one of these behind the scenes.

File: temba/channels/migrations/0021_auto_20150803_1857.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+              END;
+            $$ LANGUAGE plpgsql;
+
+            CREATE OR REPLACE FUNCTION temba_maybe_squash_channelcount(_channel_id INTEGER, _count_type VARCHAR(2), _count_day DATE) RETURNS VOID AS $$
+              DECLARE
+               current_count INT;
+              BEGIN
+                IF RANDOM() < .01 THEN
+                  IF _count_day IS NULL THEN
+                    WITH D as (DELETE FROM channels_channelcount
```
Comment (ericnewcomer) [inline temba/channels/migrations/0021_auto_20150803_1857.py]: with removed or with deleted or with ?? 

WITH D feels particularly yucky.

**Comment by ericnewcomer** [comment]
New trigger hotness! Looks great.

**Pull request merged**